lib/queueservice.js

```diff
@@ -0,0 +1,82 @@
+import xml2js from 'xml2js';
+import Debug from 'debug';
+import request from 'superagent-promise';
+
+let debug = Debug('taskcluster-docker-worker:queueService');
+
+export default class TaskQueue {
+  constructor(config) {
+    this.queues;
+    this.workerType = config.workerType;
+    this.provisionerId = config.provisionerId;
+    this.client = config.queue;
+    this.log = config.log;
+  }
+
+  async getQueues() {
+    if (!this.queues || (new Date(this.queues.expires)).getTime() > Date.now()) {
```

Comment:
hrm, I think I had this wrong, is should be < rather than greater than.  Basically wanted to find out if the time now is beyond the expiration.  So, in your suggestion we would want to do a refresh if within 5 minutes of the expiration, right?